pkg/core/resources/apis/mesh/dataplane_validator.go

```diff
@@ -225,7 +225,7 @@ func validateGateway(gateway *mesh_proto.Dataplane_Networking_Gateway) validator
 		var result validators.ValidationError
 		if protocol, exist := selector[mesh_proto.ProtocolTag]; exist {
 			if protocol != ProtocolTCP {
-				result.AddViolationAt(path.Key(mesh_proto.ProtocolTag), `other values than TCP are not allowed`)
+				result.AddViolationAt(path.Key(mesh_proto.ProtocolTag), fmt.Sprintf(`other values than TCP are not allowed, provided value "%s"`, protocol))
```

Comment:
```suggestion
				result.AddViolationAt(path.Key(mesh_proto.ProtocolTag), fmt.Sprintf(`other values than tcp are not allowed, provided value "%s"`, protocol))
```

If the right format is tcp maybe we should keep the log with tcp